cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -183,13 +185,22 @@ public String commitAndPush(CommitMeta commitMeta, Path fileChanged)
     RevCommit commit = getCommitCommand(commitMeta).call();
 
     PushCommand pushCommand = createCommand(git::push, sourceControlConfig, credentialsProvider);
-    pushCommand.call();
+    Iterable<PushResult> pushResults = pushCommand.call();
+    
+    for (PushResult result : pushResults) {
+      for (RemoteRefUpdate rru : result.getRemoteUpdates()) {
+        if (rru.getStatus() != RemoteRefUpdate.Status.OK && rru.getStatus() != RemoteRefUpdate.Status.UP_TO_DATE) {
+          throw new PushFailureException(String.format("Push failed for %s: %s", fileChanged, rru.getStatus()));
+        }
+      }
+    }
 
     try {
       TreeWalk walk = TreeWalk.forPath(git.getRepository(), fileChanged.toString(), commit.getTree());
       return walk.getObjectId(0).getName();
     } catch (IOException e) {
-      throw new PushFailureException(String.format("Failed to parse fileHash for application %s", fileChanged));
+      LOG.warn(String.format("Failed to get the fileHash for file: %s", fileChanged), e);
+      return null;
```

Comment:
should annotate the method as `@Nullable` and document when it can be null. Also make sure the caller is not expecting it to be non-null.

Reply:
Added `@Nullable`